python/ray/serve/_private/client.py

```diff
@@ -219,6 +220,59 @@ def _wait_for_deployment_deleted(self, name: str, timeout_s: int = 60):
         else:
             raise TimeoutError(f"Deployment {name} wasn't deleted after {timeout_s}s.")
 
+    def _wait_for_deployment_created(self, name: str, timeout_s: int = -1):
+        start = time.time()
+        while time.time() - start < timeout_s or timeout_s < 0:
```

Comment:
Question: it sounds like if `timeout_s` < 0, then we will run the loop indefinitely? Given that this is not async, would it block other process?

Reply:
Yup it would run indefinitely. This should only be called in either the user's python script through `serve.run()`, or in the ray task kicked off separately by the controller, so it shouldn't block anything we don't intend to block.